src/new_index/schema.rs

```diff
@@ -1756,6 +1764,12 @@ impl TxHistoryRow {
         bincode_util::serialize_big(&(code, full_hash(hash), height)).unwrap()
     }
 
+    // prefix representing the end of a given block (used for reverse scans)
+    fn prefix_height_end(code: u8, hash: &[u8], height: u32) -> Bytes {
+        // u16::MAX for the tx_position ensures we get all transactions at this height
+        bincode_util::serialize_big(&(code, full_hash(hash), height, u16::MAX)).unwrap()
```

Comment:
u16::MAX shouldn't be necessary. If we're seeking in reverse and we hit anything with the given height, (which, since the DB is sorted, will be the highest height anyways).

That said. Having this here does not harm anything. So it's fine.

Reply:
if I understand correctly, the rocksdb seek function treats the prefix we pass in as a complete key (i.e. implicitly right-padded with null bytes), so without the u16::MAX this would be equivalent to `[code, hash, height, txposition=0, ...]`.

which means it would seek to the first result *less than* `[code, hash, height]`, and then continuing to iterate in descending order of height and txposition would miss that whole block (including the after_txid we were looking for).

in any case, it didn't seem to work without it 😅